### PR TITLE
Test expected failing or unhandled cases

### PR DESCRIPTION
**Pull request by ijpulidos**
This set of changes is aimed at writing test cases that are expected to fail in the protein mutation protocol, either because of limitations of the methodology or because these are simply cases that we don't support to date.

Solves #80

### CODE REVIEW

**Comment by ijpulidos** [comment]
As usual, just bear in mind that these tests will fail as of today. We will be implementing the protocol to comply to these tests eventually. Some tweaks might be needed when that happens, but the idea is that the overall shape of these tests stay the same when we implement the protein mutation protocol.

**Comment by IAlibay** [comment]
> is there a reason generally why we want to avoid

> ring growing

Handling ring breaking is something that the current hybrid topology cannot handle properly - it's a fundamental issue in the way in which we deal with bond scaling. There are some ways around this, but they would require a rather large rewrite of the hybrid topology approach and generally isn't something we want to invest a lot of our time on.

> double charge changes

This is related to https://github.com/OpenFreeEnergy/openfe/issues/660

The main answer here is "the tooling doesn't allow it now, we could do it, but it's not the best use of our resources".

File: feflow/utils/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+"""Module with custom exceptions that can be useful for including in protocols"""
+
+
+class ProtocolError(Exception):
+    """Base exception for OpenFE protocols"""
+
+
+class MethodConstraintError(ProtocolError):
```
Comment (IAlibay) [inline feflow/utils/exceptions.py:8]: This name is a bit unintuitive imho, it makes me think of "constraints".

File: feflow/utils/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+"""Module with custom exceptions that can be useful for including in protocols"""
+
+
+class ProtocolError(Exception):
```
Comment (IAlibay) [inline feflow/utils/exceptions.py:4]: If you want to create a base error type, this really should be in gufe

File: feflow/utils/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    """Base exception for OpenFE protocols"""
+
+
+class MethodConstraintError(ProtocolError):
+    """Custom exception to be raised when a fundamental limitation in the methodology is
+    that does not support."""
+
+
+class NotSupportedError(ProtocolError):
```
Comment (IAlibay) [inline feflow/utils/exceptions.py:13]: This name is somewhat unspecific for something that inherits from something Protocol specific, could there be a better name?

**Review by IAlibay — COMMENTED** [review]
I'm not super convinced on the error names, might be something we need to workshop in the longer run.

File: feflow/tests/test_protein_mutation.py [hunk 4]
```diff
@@ -476,3 +525,71 @@ def test_charge_changing_convergence(
...
+        ala_to_pro_mapping : LigandAtomMapping
+            Mapping object representing the atom mapping from ALA to PRO.
+        """
+        from feflow.utils.exceptions import MethodConstraintError
+
+        settings = ProteinMutationProtocol.default_settings()
+        protocol = ProteinMutationProtocol(settings=settings)
+
+        # Expect an error when trying to create the DAG with this invalid transformation
+        with pytest.raises(MethodConstraintError, match="proline.*not supported"):
```
Comment (jthorton) [inline feflow/tests/test_protein_mutation.py:554]: I am unsure if it's a valid use case, but do you intend to stop all mutations involving proline? So, a proline-to-proline mutation would not be allowed in very different conformations as well?

**Comment by ijpulidos** [comment]
I renamed some of the custom errors and rewrote some of the strings, I think  we could probably reiterate on these if needed in the future. I'll go ahead and merge this.

**Pull request merged**